src/Carousel.php

```diff
@@ -20,244 +23,579 @@
+                    ->addAttributes(['aria-hidden' => 'true'])
+                    ->addClass(self::CLASS_CAROUSEL_CONTROL_PREV_ICON)
+                    ->render(),
+                "\n",
+                Span::tag()->addClass('visually-hidden')->addContent($this->controlPreviousLabel)->render(),
+                "\n",
+            )
+            ->encode(false)
+            ->render();
+    }
+
+    /**
+     * Renders the carousel indicators.
+     *
+     * @param int $key The key of the carousel item.
+     * @param CarouselItem $carouselItem The carousel item to render.
+     * @param string $id The ID of the carousel component.
+     * @param bool $active Whether the carousel item is active or not.
      *
-     * @return string the rendering result.
+     * @return string The HTML representation of the element.
      */
-    private function renderItem(array|string $item, int $index): string
+    private function renderIndicator(int $key, CarouselItem $carouselItem, string $id, bool $active): string
     {
-        if (is_string($item)) {
-            $content = $item;
-            $caption = null;
-            $options = [];
-        } elseif (isset($item['content'])) {
-            $content = $item['content'];
-            $caption = ArrayHelper::getValue($item, 'caption');
-
-            if ($caption !== null) {
-                $captionOptions = ArrayHelper::remove($item, 'captionOptions', []);
-                Html::addCssClass($captionOptions, ['captionOptions' => 'carousel-caption']);
-
-                $caption = Html::div($caption, $captionOptions)
-                    ->encode($this->encodeTags)
-                    ->render();
-            }
+        return Button::button('')
+            ->addAttributes(
+                [
+                    'data-bs-target' => '#' . $id,
+                    'data-bs-slide-to' => (string) $key,
+                    'aria-current' => $carouselItem->isActive() || $active ? 'true' : null,
+                    'aria-label' => 'Slide ' . (string) ($key + 1),
+                ],
+            )
+            ->addClass($carouselItem->isActive() || $active ? 'active' : null)
+            ->encode(false)
+            ->render();
+    }
 
-            $options = ArrayHelper::getValue($item, 'options', []);
-        } else {
-            throw new RuntimeException('The "content" option is required.');
+    /**
+     * Renders the carousel items.
+     *
+     * @param string $id The ID of the carousel component.
+     *
+     * @return string The HTML representation of the element.
+     */
+    private function renderItems(string $id): string
+    {
+        $items = [];
+        $indicators = [];
+        $renderIndicators = '';
+
+        $activeItems = array_filter($this->items, static fn (CarouselItem $item) => $item->isActive());
+
+        if (count($activeItems) > 1) {
+            throw new InvalidArgumentException('Only one carousel item can be active at a time.');
         }
```

Comment:
I thought about it too, but i need to know before iterating if any element is active, otherwise i mark the first one as active, and there may be the possibility that any element is active.